CONTRIBUTING.md

```diff
@@ -129,6 +129,21 @@ When you have set up your fork of a repository that you want to contribute to an
 4. **Validating and verifying the PR:** 
    - Reviewers can review the pull request on the GitHub website or fetch the PR using the _Fetch GitHub PR_ menu entry in the _Git Repositories_ view inside the Eclipse IDE.
    - Every PR is automatically verified to check that contributors have a [valid Eclipse Contributor Agreement (ECA)](#setting-up-your-eclipse-and-github-account).
+   ### What does a valid PR look like? (Checklist)
+   These are the basic criteria that need to be fulfilled for a PR to be considered valid. 
+   
+   **If you are a contributor, consider these points as prerequisites before your PR can be reviewed. Please ask for help if needed, either by making a comment on the PR or via the [dev mailing list](#contact).**
+
+   - All checks need to pass.
+     - If tests are broken then they should either be fixed or documented (create a new issue or mention an existing one).
+     - The code must compile.
```

Comment:
What about (new) warnings?

Reply:
Good point @laeubi, I hadn't thought of that. 

How about:

```
* There should be no new API warnings. If there are any, they should be well-grounded _e,g._ if 
there are no suitable alternatives or if implementing the alternative is outside the scope
of the current PR (in which case, a separate PR could be proposed).
```
?

Please notice that I don't say anything about existing warnings. I personally think it's acceptable to preserve the current status of the code regarding warnings, I just don't like it when it becomes worse than before.

Reply:
I think one should either fix or suppress warnings (with a comment why it can't be fixed for example).

Reply:
I added: `   - There should be no new API warnings. If a warning can’t be fixed then suppress it and write a comment explaining why.` in 8a47991e3f40f1f3fdb8cfd3ba29b9c5e5682fa8